### PR TITLE
Use Fused grouped topk for amx

### PR DESCRIPTION
**Pull request by yanbing-j**
## Motivation

This PR is to use Fused grouped topk for amx.

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/layers/moe/topk.py [hunk 1]
```diff
@@ -177,14 +180,31 @@ def select_experts(
...
         if correction_bias is None:
-            topk_weights, topk_ids = grouped_topk(
-                hidden_states=hidden_states,
-                gating_output=router_logits,
-                topk=top_k,
-                renormalize=renormalize,
-                num_expert_group=num_expert_group,
-                topk_group=topk_group,
-            )
+            if cpu_has_amx_support():
```
Comment (mingfeima) [inline python/sglang/srt/layers/moe/topk.py:183]: check device here.

**Review by mingfeima — CHANGES_REQUESTED** [review]

File: python/sglang/srt/layers/moe/topk.py (hunk 1 above)
Comment (yanbing-j) [reply python/sglang/srt/layers/moe/topk.py]: Done.

**Review by mingfeima — APPROVED** [review]
